group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroup.java

```diff
@@ -0,0 +1,1180 @@
+
+        for (String subtopologyId : allSubtopologyIds) {
+            Set<Integer> oldPartitions = oldTargetAssignment.activeTasks().getOrDefault(subtopologyId, Collections.emptySet());
+            Set<Integer> newPartitions = newTargetAssignment.activeTasks().getOrDefault(subtopologyId, Collections.emptySet());
+
+            TimelineHashMap<Integer, String> taskPartitionAssignment = invertedTargetAssignment.computeIfAbsent(
+                subtopologyId, k -> new TimelineHashMap<>(snapshotRegistry, Math.max(oldPartitions.size(), newPartitions.size()))
+            );
+
+            // Remove partitions that aren't present in the new assignment only if the partition is currently
+            // still assigned to the member in question.
+            // If p0 was moved from A to B, and the target assignment map was updated for B first, we don't want to
+            // remove the key p0 from the inverted map and undo the action when A eventually tries to update its assignment.
+            for (Integer partition : oldPartitions) {
+                if (!newPartitions.contains(partition) && memberId.equals(taskPartitionAssignment.get(partition))) {
+                    taskPartitionAssignment.remove(partition);
+                }
+            }
+
+            // Add partitions that are in the new assignment but not in the old assignment.
+            for (Integer partition : newPartitions) {
+                if (!oldPartitions.contains(partition)) {
+                    taskPartitionAssignment.put(partition, memberId);
+                }
+            }
+
+            if (taskPartitionAssignment.isEmpty()) {
+                invertedTargetAssignment.remove(subtopologyId);
+            } else {
+                invertedTargetAssignment.put(subtopologyId, taskPartitionAssignment);
+            }
+        }
+    }
+
+    /**
+     * Removes the target assignment of a member.
+     *
+     * @param memberId The member id.
+     */
+    public void removeTargetAssignment(String memberId) {
+        updateInvertedTargetActiveTasksAssignment(
+            memberId,
+            targetAssignment.getOrDefault(memberId, TasksTuple.EMPTY),
+            TasksTuple.EMPTY
+        );
+        updateInvertedTargetStandbyTasksAssignment(
+            memberId,
+            targetAssignment.getOrDefault(memberId, TasksTuple.EMPTY),
+            TasksTuple.EMPTY
+        );
+        updateInvertedTargetWarmupTasksAssignment(
+            memberId,
+            targetAssignment.getOrDefault(memberId, TasksTuple.EMPTY),
+            TasksTuple.EMPTY
+        );
+        targetAssignment.remove(memberId);
+    }
+
+    /**
+     * @return An immutable Map containing all the target assignment keyed by member id.
+     */
+    public Map<String, TasksTuple> targetAssignment() {
+        return Collections.unmodifiableMap(targetAssignment);
+    }
+
+    /**
+     * Returns the current processId of a task or null if the task does not have one.
```

Comment:
```suggestion
     * Returns the current process ID of a task or null if the task does not have one.
```
Here and in other places.

Reply:
Done